LongWarOfTheChosen/Src/LW_Toolbox_Integrated/Classes/XComGameState_Unit_LWRandomizedStats.uc

```diff
@@ -99,26 +99,32 @@ function RandomizeInitialStats(XComGameState_Unit Unit)
 		ComIntTotalWeight += ComIntSwap.Weight;
 	}
 
+	//Roll a chance for a stat swap and don't roll combat intelligence for rebels
+	if (`SYNC_RAND(100) < default.COMINT_BASE_SWAP_CHANCE && InStr(Unit.GetMyTemplateName(), "Rebel") != 0 )
+	{
+		do 
+		{
+			ComIntSwap = SelectComIntStatSwap(ComIntTotalWeight);
+		} until (IsValidComIntSwap(ComIntSwap, Unit) || (++iterations > 1000));
```

Comment:
If we're going to put this first, we should reduce the number of potential iterations, otherwise a failure here will block any other stat swaps. Alternatively, reset `iterations` between the loops.